sds_data_manager/sds_data_manager_stack.py

```diff
@@ -15,7 +15,13 @@
 
 class SdsDataManagerStack(Stack):
     def __init__(
-        self, scope: Construct, construct_id: str, sds_id: str, **kwargs
+        self,
+        scope: Construct,
+        construct_id: str,
+        sds_id: str,
+        userpool_id: str = "",
+        app_client_id: str = "",
```

Comment:
```suggestion
        userpool_id: str
        app_client_id: str,
```
Can the ID be an empty string? Perhaps push the validation up a higher level and make this an argument without a default.